### ISSUE TITLE
OpenChecklist - cannot tick the last checklist item

### ISSUE THREAD

**Issue by psysoo** [GuideBlocks-org/iOS-GuideBlocks#57]
OpenChecklist - cannot tick the last checklist item

https://github.com/GuideBlocks-org/iOS-GuideBlocks/assets/114545794/17b51844-f205-4f73-aaab-2acc58bf19d6

**Comment by amraboelela** [comment]
It works with me for visiting a specific screen.
https://github.com/GuideBlocks-org/iOS-GuideBlocks/assets/882954/93ceeef2-c611-4822-b65b-66d043995e68

**Comment by amraboelela** [comment]
This task will be done only when `favourited_properties_count` is 3 or more, maybe i shouldn't make it clickable?
        {
            "name": "Check User Favourites Count",
            "action": "checkTag",
            "action_data": {
                "key": "favourited_properties_count",
                "value": "3",
                "type": "int",
                "operator": "gte" # greater than or equals
            }
        }

**Comment by djinoz** [comment]
I agree Amr, we need a test case to cause the tag to happen, how did you test it and how can QA test?

**Comment by psysoo** [comment]
@amraboelela I just copied sample yml file and pasted it. then click the 4th item when guide is shown up. 

I was trying to test to set the tag and value 3 by the first checklist item. BUT it's blocked by the following bug; somehow shows 8 more tasks instead of showing the items in extra_json. 

<img width="1680" alt="Screenshot 2024-03-04 at 4 38 51 pm" src="https://github.com/GuideBlocks-org/iOS-GuideBlocks/assets/114545794/9bace64f-c637-4ca7-8769-42056ae4a779">

**Comment by amraboelela** [comment]
I pasted the json from yaml file as:

{
    "guideBlockKey": "OpenChecklist",
    "tasks": [
        {
            "name": "Set My Tag",
            "action": "setTag",
            "action_data": {
                "key": "mytag",
                "value": "1234"
            }
        },
        {
            "name": "Visit Inbox",
            "action": "gotoScreen",
            "action_data": {
                "deep_link": "airbnbContextual://tab/inbox"
            }
        },
        {
            "name": "Visit Profile",
            "action": "gotoScreen",
            "action_data": {
                "deep_link": "airbnbContextual://tab/profile"
            }
        },
        {
            "name": "Check User Favourites Count",
            "action": "checkTag",
            "action_data": {
                "key": "favourited_properties_count",
                "value": "3",
                "type": "int",
                "operator": "gte"
            }
        }
    ]
}
And got the following:

<img width="1069" alt="Screenshot 2024-03-04 at 6 46 45 PM" src="https://github.com/GuideBlocks-org/iOS-GuideBlocks/assets/882954/479ec4b0-9c94-4d64-a53e-456a41329f74">

**Comment by amraboelela** [comment]
Try again after merging #73 now clicking on the 4th row doesn't make the list to close.
<img width="1313" alt="Screenshot 2024-03-04 at 7 16 06 PM" src="https://github.com/GuideBlocks-org/iOS-GuideBlocks/assets/882954/c63946df-93b5-4dc6-98f7-ab227a1245f3">

**Comment by amraboelela** [comment]
> I agree Amr, we need a test case to cause the tag to happen, how did you test it and how can QA test?

I tested it by setting the tag through the code, the QA could test it if he can set the tag manually somehow.
